core/src/main/java/hudson/model/Job.java

```diff
@@ -221,29 +221,20 @@ public void onLoad(ItemGroup<? extends Item> parent, String name)
 
         TextFile f = getNextBuildNumberFile();
         if (f.exists()) {
-            // starting 1.28, we store nextBuildNumber in a separate file.
-            // but old Hudson didn't do it, so if the file doesn't exist,
-            // assume that nextBuildNumber was read from config.xml
             try {
                 synchronized (this) {
                     this.nextBuildNumber = Integer.parseInt(f.readTrim());
                 }
             } catch (NumberFormatException e) {
                 LOGGER.log(Level.WARNING, "Corruption in {0}: {1}", new Object[] {f, e});
-                //noinspection StatementWithEmptyBody
-                if (this instanceof LazyBuildMixIn.LazyLoadingJob) {
-                    // allow LazyBuildMixIn.onLoad to fix it
-                } else {
-                    RunT lB = getLastBuild();
-                    synchronized (this) {
-                        this.nextBuildNumber = lB != null ? lB.getNumber() + 1 : 1;
-                    }
-                    saveNextBuildNumber();
+                RunT lB = getLastBuild();
+                synchronized (this) {
+                    this.nextBuildNumber = lB != null ? lB.getNumber() + 1 : 1;
                 }
+                saveNextBuildNumber();
             }
-        } else {
-            // From the old Hudson, or doCreateItem. Create this file now.
```

Comment:
In fact `doCreateItem` calls `onCreatedFromScratch`, which did _not_ create this file, so that was misleading.